src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let sortingFunc;
+
+    // Compare elements as strings if compareFunction is omitted
+    if (compareFunction !== undefined) {
+      sortingFunc = compareFunction;
+    } else {
+      sortingFunc = (a, b) => String(a) > String(b);
+    }
+
+    // Iterate through the array {this.length - 1} amount of times,
+    // skipping the sorted values
+    for (let maxLength = this.length; maxLength > 0; maxLength--) {
+      for (let i = 1; i < maxLength; i++) {
+        const a = this[i - 1];
+        const b = this[i];
+        const comparissonResult = sortingFunc(a, b);
```

Comment:
The variable name 'comparissonResult' contains a typo. It should be 'comparisonResult'.